regolith/helpers/l_contactshelper.py

```diff
@@ -45,11 +46,23 @@ def subparser(subpi):
         "--notes",
         help='fragment (single argument only) to be found in the notes section of a contact'
     )
+    subpi.add_argument(
+        "-f",
+        "--filter",
+        nargs="+",
+        help='Search this collection by giving key element pairs'
+    )
+    subpi.add_argument(
+        "-k",
+        "--keys",
+        nargs="+",
+        help='Specify what keys to return when running --filter. If no argument is given the default is just the id.'
+    )
     return subpi
 
 
 def stringify(con):
```

Comment:
I dislike this stringify function tbh, I think it is a bit brutish and have a feeling there is a better way.  Basically, the retrieval functions return documents, which are rich in information, and the stringify blows away most of the info in a very specific way.  It is much better to hold on to the filtered documents and propagate them, if you can figure out how to get the desired behavior.  I think it was Shomik who did this and at some point I wanted to get his thing merged and it was working, so I let it pass, but can you look at it?